### PR TITLE
[REBASED] Apply Transform Workflow

### PR DESCRIPTION
**Pull request by skoudoro**
The goal of this PR is to clean and rebase #1605. Look at #1605 for more information

### CODE REVIEW

File: dipy/io/image.py [hunk 1]
```diff
@@ -27,6 +27,16 @@ def save_nifti(fname, data, affine, hdr=None):
     result_img.to_filename(fname)
 
 
+def load_affine_matrix(fname):
```
Comment (arokem) [inline dipy/io/image.py]: This is just a wrapper around `np.loadtxt`. Let's use that one.

Comment (skoudoro) [reply dipy/io/image.py]: thank you! addressed in d90ea0d

File: dipy/io/image.py [hunk 2]
```diff
@@ -27,16 +27,6 @@ def save_nifti(fname, data, affine, hdr=None):
...
-    """
-        Parameters
-        ---------
-        fname : str
-            The file containing the saved affine matrix.
-    """
-    return np.loadtxt(fname)
-
-
 def save_affine_matrix(fname, affine):
```
Comment (arokem) [inline dipy/io/image.py]: While we're here, could you please get rid of `save_affine_matrix`? That looks like a wrapper around `np.savetxt`.

Comment (skoudoro) [reply dipy/io/image.py]: ok, done in 4306c36

**Comment by arokem** [comment]
LGTM. Will merge in a couple of days, unless someone has something to add.

**Comment by skoudoro** [comment]
This PR is ready to go! Does anyone else want to take a look?

**Pull request merged**